### PR TITLE
DEP Move solr-php-client with PHP 8.1 compatbility to thirdparty folder

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10250

I attempted to [update this](https://github.com/PTCInc/solr-php-client/pull/8) upstream but got no response.  Looks abandoned.  So just adding to thirdparty folder instead.

The [COPYING.md](https://github.com/PTCInc/solr-php-client/blob/master/COPYING) file says it's ok to copy this, as long as copyright is retained

Includes a fix to add method attribute to `Document::getIterator()` for PHP 8.1 compatibility

### CODE REVIEW

**Comment by dhensby** [comment]
Potentially time to migrate to https://github.com/solariumphp/solarium ? I think I looked into this quite some time ago, not sure what happened to it, though.

**Comment by emteknetnz** [comment]
That seems like substantially more work.  There was only a single line required to get this php 8.1 compatible (at least according to unit tests).

File: src/Search/Processors/SearchUpdateProcessor.php [hunk 1]
```diff
@@ -154,6 +155,9 @@ protected function getSource()
      */
     public function process()
     {
+        if (!DB::is_active()) {
```
Comment (GuySartorelli) [inline src/Search/Processors/SearchUpdateProcessor.php:158]: What prompted this? It doesn't seem related to the rest of the PR?

**Review by GuySartorelli — COMMENTED** [review]
I've checked, and the only difference between thirdparty/solr-php-client and the v1.0.0 tag of the package in github is the change made in the PR referenced in the description.

This module honestly feels a bit legacy at this point - personally I'm okay with taking the less-effort approach in this case, but given there may be some controversy I'll leave it for someone else to merge.

File: src/Search/Processors/SearchUpdateProcessor.php (hunk 1 above)
Comment (emteknetnz) [reply src/Search/Processors/SearchUpdateProcessor.php:158]: PR that went into 3.9 though wasn't merged up - https://github.com/silverstripe/silverstripe-fulltextsearch/pull/318

**Comment by dhensby** [comment]
Yeah, I think that moving to solarium should be a long-term project and not a way to solve php 8.1 compatibility. We should probably aim not to be reliant on libraries that are abandoned or haven't seen any effort in 6 years!

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
My preference would be to do a `sminee/phpunit` and fork the repo on the Silverstripe domain and stick a big nasty warning at the top that it's not supported. But this solution could work as well.

FYI Silverstripe Ltd has moved on to using Elastic for search in its new project. We'll keep maintaining this module because a lot of people are still using it, but it's unlikely we'll update it to work with the later version of Solr.

File: composer.json [hunk 2]
```diff
@@ -24,7 +24,6 @@
         "php": "^7.4 || ^8.0",
         "silverstripe/framework": "^4.10",
         "monolog/monolog": "~1.15",
-        "ptcinc/solr-php-client": "^1.0",
```
[the remark this answers is not in the document]
Comment (emteknetnz) [reply composer.json:27]: I don't have sufficient permissions to fork to the silverstripe account.  If you'd like to go this route could you please fork it.

[the remark this answers is not in the document]
Comment (dhensby) [reply composer.json:27]: fork here: https://github.com/silverstripe/solr-php-client - I can delete it if we decide we don't want it

**Comment by GuySartorelli** [comment]
[FYI](https://github.com/PTCInc/solr-php-client/issues/9#issuecomment-1106463063)
> I was the primary author. This was created for a product which no longer exists, but was acquired by PTC. There is unlikely to be any changes here as I am no longer part of PTC. Anyone still using this probably forks to publish or vendor their own version. Sorry for your inconvenience.

**Comment by emteknetnz** [comment]
Thanks for forking the module @dhensby - we'll use that.  I've created a [follow up PR](https://github.com/silverstripe/solr-php-client/pull/1) to fix a couple of things on the fork
